src/VisualStudio/Core/Def/Implementation/Diagnostics/MiscellaneousDiagnosticAnalyzerService.cs

```diff
@@ -82,7 +90,9 @@ public SyntaxOnlyDiagnosticAnalyzer(MiscellaneousDiagnosticAnalyzerService servi
                 Contract.Requires(document.Project.Solution.Workspace == _workspace);
 
                 var diagnosticData = diagnostics == null ? ImmutableArray<DiagnosticData>.Empty : diagnostics.Select(d => DiagnosticData.Create(document, d)).ToImmutableArrayOrEmpty();
-                _service.RaiseDiagnosticsUpdated(new DiagnosticsUpdatedArgs(new MiscUpdateArgsId(document.Id), _workspace, document.Project.Solution, document.Project.Id, document.Id, diagnosticData));
+                _service.RaiseDiagnosticsUpdated(
+                    new DiagnosticsUpdatedArgs(new MiscUpdateArgsId(document.Id, _service.analyzerService.GetCompilerDiagnosticAnalyzer(document.Project.Language)),
+                    _workspace, document.Project.Solution, document.Project.Id, document.Id, diagnosticData));
```

Comment:
It will probably not be useful to the user - however, it may be useful for us to debug stale error issues... I am fine with adding such a suffix i.e. something like - "VB Compiler [Live Analysis]"

For errors that come from command line build we call them "Build". Will we change this to "VB Compiler [Build]" (or simply "VB Compiler") to match or will that continue to be just "Build"?